x-pack/plugins/security_solution/server/lib/detection_engine/rule_response_actions/endpoint_response_action.ts

```diff
@@ -5,26 +5,46 @@
  * 2.0.
  */
 
-import { each } from 'lodash';
+import { each, map, uniq } from 'lodash';
+import { ALERT_RULE_NAME, ALERT_RULE_UUID } from '@kbn/rule-data-utils';
+import type { ResponseActionAlerts } from './types';
 import type { EndpointAppContextService } from '../../../endpoint/endpoint_app_context_services';
 import type { RuleResponseEndpointAction } from '../../../../common/detection_engine/rule_response_actions/schemas';
-import type { AlertsWithAgentType } from './types';
 
 export const endpointResponseAction = (
   responseAction: RuleResponseEndpointAction,
   endpointAppContextService: EndpointAppContextService,
-  { alertIds, agentIds, ruleId, ruleName, hosts }: AlertsWithAgentType
-) =>
-  Promise.all(
+  { alerts: filteredAlerts }: ResponseActionAlerts
+) => {
+  const { comment, command } = responseAction.params;
+  const commonData = {
+    comment,
+    command,
+    rule_id: filteredAlerts[0][ALERT_RULE_UUID],
+    rule_name: filteredAlerts[0][ALERT_RULE_NAME],
+  };
+  const agentIds = uniq(map(filteredAlerts, 'agent.id'));
+  const alertIds = map(filteredAlerts, '_id');
+
+  const hosts = filteredAlerts.reduce((acc: Record<string, string>, alert) => {
```

Comment:
Don't have to change but it is easier to read a reduce function if you type it this way.
```suggestion
  const hosts = filteredAlerts.reduce<Record<string, string>>((acc, alert) => {
```